helix-term/src/commands.rs

```diff
@@ -3284,33 +3286,88 @@ fn select_register(cx: &mut Context) {
     })
 }
 
-fn space_mode(cx: &mut Context) {
-    cx.on_next_key(move |cx, event| {
-        if let KeyEvent {
-            code: KeyCode::Char(ch),
-            ..
-        } = event
-        {
-            // TODO: temporarily show SPC in the mode list
-            match ch {
-                'f' => file_picker(cx),
-                'b' => buffer_picker(cx),
-                's' => symbol_picker(cx),
-                'w' => window_mode(cx),
-                'y' => yank_joined_to_clipboard(cx),
-                'Y' => yank_main_selection_to_clipboard(cx),
-                'p' => paste_clipboard_after(cx),
-                'P' => paste_clipboard_before(cx),
-                'R' => replace_selections_with_clipboard(cx),
-                // ' ' => toggle_alternate_buffer(cx),
-                // TODO: temporary since space mode took its old key
-                ' ' => keep_primary_selection(cx),
-                _ => (),
-            }
+macro_rules! mode_info {
+    // TODO: how to use one expr for both pat and expr?
```

Comment:
Here how you could fix this TODO: https://github.com/helix-editor/helix/commit/6a52a3a9328c6374d0b2988d3412e79e8d546528
Feel free to cherry pick and squash if you want

Reply:
Maybe you can improve the mode_info macro instead by adding `ctrl $key:expr`, `alt $key:expr` patterns
I don't see many ways to do that

Reply:
I was also thinking the `key!` macro should be more flexible. `key!('a')`, `key!(CTRL 'a')` etc

Reply:
Last time I got `key!(c-'a')` working but not sure if it's a good idea, because it feels kinda messy to me.

> Maybe you can improve the mode_info macro instead by adding ctrl $key:expr, alt $key:expr patterns

Yeah, could do that but I think deduplication of the stuff available is better to be done.

Reply:
There is no problem with supporting `key!(CTRL 'a')` etc I think.

> Yeah, could do that but I think deduplication of the stuff available is better to be done.

What do you mean? Using `CTRL 'a'` and so on will duplicate code? I don't see what prevent us from refactoring with this approach

Reply:
I prefer the approach to be more similar to our current keybindings, which is to parse the keys so that we can easily port it to use config later. One issue I see with config is how can we even specify the keys for prompt or popups?

Reply:
I think it's fine if you don't care to much about the current keymap config, it's mostly a temporary option.

Reply:
I changed it to use current parsing keys.